src/frequenz/sdk/config/_config_managing.py

```diff
@@ -150,51 +183,56 @@ async def _run(self) -> None:
         """
         await self.send_config()
 
-        parent_paths = {p.parent for p in self._config_paths}
-
-        # FileWatcher can't watch for non-existing files, so we need to watch for the
-        # parent directories instead just in case a configuration file doesn't exist yet
-        # or it is deleted and recreated again.
-        file_watcher = FileWatcher(
-            paths=list(parent_paths),
-            event_types=self._event_types,
-            force_polling=self._force_polling,
-            polling_interval=self._polling_interval,
-        )
-
-        try:
-            async for event in file_watcher:
-                # Since we are watching the whole parent directories, we need to make
-                # sure we only react to events related to the configuration files we
-                # are interested in.
-                if not any(event.path.samefile(p) for p in self._config_paths):
-                    continue
-
-                match event.type:
-                    case EventType.CREATE:
-                        _logger.info(
-                            "%s: The configuration file %s was created, sending new config...",
-                            self,
-                            event.path,
-                        )
-                        await self.send_config()
-                    case EventType.MODIFY:
-                        _logger.info(
-                            "%s: The configuration file %s was modified, sending update...",
-                            self,
-                            event.path,
-                        )
-                        await self.send_config()
-                    case EventType.DELETE:
-                        _logger.info(
-                            "%s: The configuration file %s was deleted, ignoring...",
-                            self,
-                            event.path,
-                        )
-                    case _:
-                        assert_never(event.type)
-        finally:
-            del file_watcher
+        async for event in self._file_watcher:
+            if not event.path.exists():
+                _logger.error(
+                    "%s: Received event %s, but the watched path %s doesn't exist.",
+                    self,
+                    event,
+                    event.path,
+                )
+                continue
+            # Since we are watching the whole parent directories, we need to make
+            # sure we only react to events related to the configuration files we
+            # are interested in.
+            #
+            # pathlib.Path.samefile raises error if any path doesn't exist so we need to
+            # make sure `event.path` and `p` exists before calling it.
+            #
+            # On EventType.DELETE event the event.path won't exists and we have no way
+            # to check if this path is in self._config_paths so we need to skip this
+            # event.
```

Comment:
This now looks outdated, maybe something like:
```suggestion
            # pathlib.Path.samefile raises error if any path doesn't exist so we need to
            # make sure the paths exists before calling it. This could happen as it is not
            # required that all config files exist, only one is required but we don't know which.
```

Reply:
fixed